src/cloudsc_python/drivers/config.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+from __future__ import annotations
+import numpy as np
+from os.path import dirname, join, normpath, splitext
+from pydantic import BaseModel, validator
+import socket
+from typing import Optional
+
+from cloudsc4py.framework.config import DataTypes, GT4PyConfig
+
+
+class IOConfig(BaseModel):
+    output_file: Optional[str]
+    host_name: Optional[str]
+
+    @validator("output_file")
+    def check_extension(cls, v: Optional[str]) -> Optional[str]:
+        if v is None:
+            return v
+
+        basename, extension = splitext(v)
+        if extension == "":
+            return v + ".csv"
+        elif extension == ".csv":
+            return v
+        else:
+            return basename + ".csv"
+
+    @validator("host_name")
+    def set_host_name(cls, v: Optional[str]) -> str:
+        return v or socket.gethostname()
+
+    def with_host_name(self, host_name: str) -> IOConfig:
+        args = self.dict()
+        args["host_name"] = host_name
+        return IOConfig(**args)
+
+    def with_output_file(self, output_file: str) -> IOConfig:
+        args = self.dict()
+        args["output_file"] = output_file
+        return IOConfig(**args)
+
+
+default_io_config = IOConfig(output_file=None, host_name=None)
+
+
+class PythonConfig(BaseModel):
+    # domain
+    nx: Optional[int]
```

Comment:
I've added a few more comments here and there to improve comprehension. Plus, I renamed `nx` as `num_cols` for better clarity. I opted for `num_cols` rather than `ncol` to be consistent with `num_cols` and `num_threads`.